lib/espree.js

```diff
@@ -36,11 +38,39 @@ function normalizeEcmaVersion(ecmaVersion) {
                 return version;
 
             default:
-                throw new Error("Invalid ecmaVersion.");
         }
-    } else {
-        return DEFAULT_ECMA_VERSION;
     }
+
+    throw new Error("Invalid ecmaVersion.");
+}
+
+/**
+ * Normalize sourceType from the initial config
+ * @param {string} sourceType to normalize
+ * @throws {Error} throw an error if sourceType is invalid
+ * @returns {string} normalized sourceType
+ */
+function normalizeSourceType(sourceType = "script") {
+    if (sourceType === "script" || sourceType === "module") {
+        return sourceType;
+    }
+    throw new Error("Invalid sourceType.");
+}
+
+/**
+ * Normalize parserOptions
+ * @param {Object} options the parser options to normalize
+ * @throws {Error} throw an error if found invalid option.
+ * @returns {Object} normalized options
+ */
+function normalizeOptions(options) {
+    const ecmaVersion = normalizeEcmaVersion(options.ecmaVersion);
+    const sourceType = normalizeSourceType(options.sourceType);
+
+    if (sourceType === "module" && ecmaVersion < 6) {
+        throw new Error("sourceType 'module' is not supported when ecmaVersion < 6!");
```

Comment:
I don't think we should use an exclamation point here-- it's unnecessary, and could potentially confuse people.

```suggestion
        throw new Error("sourceType 'module' is not supported when ecmaVersion < 6");
```